modules/sync/syncs/httpsynchronization.cpp

```diff
@@ -97,24 +98,47 @@ void HttpSynchronization::start() {
     }
     _state = State::Syncing;
 
-    if (hasSyncFile()) {
+    if (checkSyncFile()) {
         _state = State::Resolved;
         return;
     }
 
+    if (isRejected()) {
+        return;
+    }
+
     std::string query = fmt::format(
         "?identifier={}&file_version={}&application_version={}",
         _identifier, _version, ApplicationVersion
     );
 
     _syncThread = std::thread(
         [this](const std::string& q) {
-            for (const std::string& url : _syncRepositories) {
-                const bool success = trySyncFromUrl(url + q);
-                if (success) {
-                    createSyncFile();
-                    _state = State::Resolved;
-                    return;
+        for (const std::string& url : _syncRepositories) {
+            //TODO: handle multiple syncRepositories, before it would only create
+            // syncfile from first success and then continue
+            const bool success = trySyncFromUrl(url + q);
+            if (success) {
+                _state = State::Resolved;
+                createSyncFile(success);
+                return;
```

Comment:
Probably better to just do a `break` in here.  It might be not obvious to other people that you are hard ejecting from the middle of a for loop and there might be code added later after the for loop that we'd want to run even when we successfully synced a file